src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java

```diff
@@ -148,15 +148,13 @@ public QueryByUIFilterGroup(IForm form, FilterGroup filterGroup, CompletionFlag
             super.addFilter(fem, op, compareValue);
             break;
           case JRDATETIME:
-            if (fem.getFormDataModel() == null && fem.getElementName().contains("meta-"))
+          case JRDATE:
+          case JRTIME:
+            if (isLegacyTemporalField(fem))
               super.addFilter(fem, op, compareValue);
-            else if (fem.getFormDataModel() != null)
-              super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
             else
-              throw new IllegalArgumentException("Unrecognized dateTime field");
+              super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
```

Comment:
I've removed the exception because now I know more about this and I'm more confident about the range of scenarios we could get.